test/jdk/sun/java2d/Disposer/TestDisposerRace.java

```diff
@@ -150,6 +152,7 @@ private static void giveGCAChance() {
                 Thread.sleep(2000); // Give GC a little chance to run
                 break;
             } catch (OutOfMemoryError | InterruptedException ignored2) {
+                System.gc(); // If we can't even sleep, ask the GC for assistance
                 continue;
```

Comment:
@dholmes-ora @AlanBateman @DougLea With this addition I managed to get the TestDisposerRace.java to complete successfull over 100.000 times.